ethereum/json-types/src/main/java/tech/pegasys/teku/ethereum/json/types/node/PeerCountBuilder.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Consensys Software Inc., 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.ethereum.json.types.node;
+
+import static tech.pegasys.teku.infrastructure.json.types.CoreTypes.UINT64_TYPE;
+
+import java.util.function.Function;
+import tech.pegasys.teku.infrastructure.json.types.DeserializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+
+public class PeerCountBuilder {
+
+  public static final DeserializableTypeDefinition<PeerCount> PEER_COUNT_DATA_TYPE =
+      DeserializableTypeDefinition.object(PeerCount.class, PeerCountBuilder.class)
+          .initializer(PeerCountBuilder::new)
+          .finisher(PeerCountBuilder::build)
+          .withField(
+              "disconnected",
+              UINT64_TYPE,
+              PeerCount::getDisconnected,
+              PeerCountBuilder::disconnected)
+          .withField(
+              "connecting", UINT64_TYPE, PeerCount::getConnecting, PeerCountBuilder::connecting)
+          .withField("connected", UINT64_TYPE, PeerCount::getConnected, PeerCountBuilder::connected)
+          .withField(
+              "disconnecting",
+              UINT64_TYPE,
+              PeerCount::getDisconnecting,
+              PeerCountBuilder::disconnecting)
+          .build();
+
+  public static final DeserializableTypeDefinition<PeerCount> PEER_COUNT_TYPE =
+      DeserializableTypeDefinition.object(PeerCount.class, PeerCountBuilder.class)
+          .name("GetPeerCountResponse")
+          .initializer(PeerCountBuilder::new)
+          .finisher(PeerCountBuilder::build)
+          .withField("data", PEER_COUNT_DATA_TYPE, Function.identity(), PeerCountBuilder::peerCount)
+          .build();
+
+  private UInt64 disconnected;
+  private UInt64 connecting = UInt64.ZERO;
```

Comment:
Why connecting and disconnecting defaults to ZERO? all are required by spec.

Reply:
Ahh I see, we are not setting it in builder. I'd just set ZERO to all by default to avoid confusion. You cannot know all usage when creating isolated class.

Reply:
Actually we used to set `connecting` and `disconnecting` to 0 because we don't have that information in the `LibP2PPeer`... But I agree that it's better to avoid confusion, updated it to initialise all the values to 0